hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestReconWithOzoneManagerFSO.java

```diff
@@ -162,6 +199,104 @@ public void testNamespaceSummaryAPI() throws Exception {
     Assert.assertEquals(12, rootBasicEntity.getCountStats().getNumTotalKey());
   }
 
+
+  /**
+   * Test case for verifying directory deletion and namespace summary updates.
+   * Three cases are tested:
+   * CASE-1: Creating a directory structure with files and verifying OM and
+   *          Recon tables.
+   * CASE-2: Deleting files from the directory and verifying updated NS summary.
+   * CASE-3: Deleting the entire directory and confirming the NS summary.
+   */
+  @Test
+  public void testNamespaceSummaryUpdatesForDirectoryDeletion()
+      throws Exception {
+
+    // CASE-1
+    // Create a directory structure with 10 files in dir1
+    addKeysToDirectory(1, 10, "/dir1");
+
+    // Fetch the file table and directory table from Ozone Manager.
+    OMMetadataManager ozoneMetadataManagerInstance =
+        cluster.getOzoneManager().getMetadataManager();
+    Table<String, OmKeyInfo> omFileTable =
+        ozoneMetadataManagerInstance.getKeyTable(getFSOBucketLayout());
+    Table<String, OmDirectoryInfo> omDirTable =
+        ozoneMetadataManagerInstance.getDirectoryTable();
+
+    // Verify the entries in the Ozone Manager tables.
+    assertOmTableRowCount(omFileTable, 10);
+    assertOmTableRowCount(omDirTable, 1);
+
+    // Sync data from Ozone Manager to Recon.
+    OzoneManagerServiceProviderImpl impl = (OzoneManagerServiceProviderImpl)
+        cluster.getReconServer().getOzoneManagerServiceProvider();
+    impl.syncDataFromOM();
+
+    // Retrieve tables from Recon's OM-DB.
+    ReconOMMetadataManager reconOmMetadataManagerInstance =
+        (ReconOMMetadataManager) cluster.getReconServer()
+            .getOzoneManagerServiceProvider().getOMMetadataManagerInstance();
+    Table<String, OmKeyInfo> reconFileTable =
+        reconOmMetadataManagerInstance.getKeyTable(getFSOBucketLayout());
+    Table<String, OmDirectoryInfo> reconDirTable =
+        reconOmMetadataManagerInstance.getDirectoryTable();
+    Table<String, OmKeyInfo> reconDeletedDirTable =
+        reconOmMetadataManagerInstance.getDeletedDirTable();
+
+    // Verify the entries in the Recon tables after sync.
+    assertReconTableRowCount(reconFileTable, 10);
+    assertReconTableRowCount(reconDirTable, 1);
+    assertReconTableRowCount(reconDeletedDirTable, 0);
+
+    // Retrieve the object ID of dir1 from directory table.
+    Long directoryObjectId = null;
+    try (
+        TableIterator<?, ? extends Table.KeyValue<?, OmDirectoryInfo>> iterator
+            = reconDirTable.iterator()) {
+      if (iterator.hasNext()) {
+        directoryObjectId = iterator.next().getValue().getObjectID();
```

Comment:
Other test cases not adding any directories in cluster ? How are we sure it will retrieve "dir1" object Id.